src/net_processing.cpp

```diff
@@ -1422,26 +1429,53 @@ void PeerManagerImpl::FindNextBlocksToDownload(const Peer& peer, unsigned int co
 
 void PeerManagerImpl::PushNodeVersion(CNode& pnode, const Peer& peer)
 {
-    uint64_t my_services{peer.m_our_services};
-    const int64_t nTime{count_seconds(GetTime<std::chrono::seconds>())};
-    uint64_t nonce = pnode.GetLocalNonce();
-    const int nNodeStartingHeight{m_best_height};
-    NodeId nodeid = pnode.GetId();
-    CAddress addr = pnode.addr;
-
-    CService addr_you = addr.IsRoutable() && !IsProxy(addr) && addr.IsAddrV1Compatible() ? addr : CService();
-    uint64_t your_services{addr.nServices};
-
-    const bool tx_relay{!RejectIncomingTxs(pnode)};
-    m_connman.PushMessage(&pnode, CNetMsgMaker(INIT_PROTO_VERSION).Make(NetMsgType::VERSION, PROTOCOL_VERSION, my_services, nTime,
-            your_services, addr_you, // Together the pre-version-31402 serialization of CAddress "addrYou" (without nTime)
-            my_services, CService(), // Together the pre-version-31402 serialization of CAddress "addrMe" (without nTime)
-            nonce, strSubVersion, nNodeStartingHeight, tx_relay));
-
+    uint64_t my_services;
+    int64_t my_time;
+    uint64_t your_services;
+    CService your_addr;
+    std::string my_user_agent;
+    int my_height;
+    bool my_tx_relay;
+    if (pnode.IsSensitiveRelayConn()) {
+        my_services = NODE_NONE;
+        my_time = 0;
+        your_services = NODE_NONE;
+        your_addr = CService{};
+        my_user_agent = "/pynode:0.0.1/";
```

Comment:
It seems for extra anonymity, but I have a similar doubt, couldn't I make easier for someone to censor us?

Reply:
unless we want to sybil the network and fake long-term connections, it's obvious what this connection will be about? Just retry with new peers until someone honest propagates it?

Reply:
TLDR: @instagibbs, right! :)

First, the problem with the user-agent string is that it can be customized which will totally reveal the identity of the sender. Next idea is to use the non-customized user-agent even if `-uacomment=` is configured. But then, why even reveal the version - whether it is 26.0 or 27.0? Right after a new release there will be few nodes that run the just-released version and thus the anonymity set will be small. So this string better be constant - either empty string or whatever other constant. `/pynode:0.0.1/` was suggested here - https://github.com/bitcoin/bitcoin/pull/27509#issuecomment-1517896418

Reply:
Ok understood. In that case, would it make sense to be very explicit about what this connection is about? I mean user agent could essentially be "I am going to send you one TX and then disconnect". Additional behavior could even be encoded on the recipient side, maybe to ensure we don't disconnect prematurely ?

Reply:
I don't like outright lying in the UA. Why not just send it blank? Or at least some kind of standard for "I am not telling you"

Reply:
_(beware, https://bikeshed.com/)_

I think any constant is ok and `bitcoin-submittx`'s string (`/pynode:0.0.1/`) was already suggested by @sipa here https://github.com/bitcoin/bitcoin/pull/27509#issuecomment-1517896418. That suggestion has 6 positive reactions. It is about increasing the anonymity set, in case different versions or different softwares use this technique.

> some kind of standard for "I am not telling you"

I don't think there is, but we can standardize `/pynode:0.0.1/` as "I am not telling you"? ;-)

Should I change this to an empty string? If this comment collects more :+1: + :heart: than https://github.com/bitcoin/bitcoin/pull/27509#issuecomment-1517896418 then I will change it.